lib/game/listeners.go

```diff
@@ -16,12 +16,18 @@ type ChunkListener interface {
 	ChunkGenerated(cc coords.Chunk, data *mapgen.Chunk)
 }
 
-type EntityListener interface {
-	EntityCreated(id EntityID, entity Entity)
-	EntityUpdated(id EntityID, entity Entity)
-	EntityDamaged(id EntityID, entity Entity)
-	EntityDied(id EntityID, entity Entity, killer string)
-	EntityRemoved(id EntityID)
+type BioticListener interface {
+	BioticCreated(id EntityId, biotic Biotic)
+	BioticUpdated(id EntityId, biotic Biotic)
+	BioticDamaged(id EntityId, biotic Biotic)
+	BioticDied(id EntityId, biotic Biotic, killer string)
+	BioticRemoved(id EntityId)
+}
+
+type WorldItemListener interface {
+	WorldItemAdded(id EntityId, worldItem *WorldItem)
+	WorldItemUpdated(id EntityId, worldItem *WorldItem)
+	WorldItemRemoved(id EntityId)
```

Comment:
I don't see the problem with having lots of listeners. I'll admit that Go makes it really annoying to have a ton of listeners, but it really helps to separate things out when the game gets more complicated.